extensions/ql-vscode/src/view/common/DataGrid.tsx

```diff
@@ -0,0 +1,79 @@
+import * as React from "react";
+import { styled } from "styled-components";
+
+/*
+ * A drop-in replacement for the VSCodeDataGrid family of components.
+ *
+ * The difference is that the `display: grid` styling is applied to `DataGrid`, whereas
+ * in the VS Code version that styling is applied to `VSCodeDataGridRow`. This gives
+ * column alignment across rows in situation with dynamic contents. It also allows
+ * for cells to span multiple rows and all the other features of data grids.
+ */
+
+const StyledDataGrid = styled.div<{ $gridTemplateColumns: string | number }>`
+  display: grid;
+  grid-template-columns: ${(props) => props.$gridTemplateColumns};
+  padding: calc((var(--design-unit) / 4) * 1px) 0;
```

Comment:
The `--design-unit` variable seems to come from FAST foundation, but since this is not a FAST foundation component I'm not sure we should be using it here. Could we simplify this to `1px`?

```suggestion
  padding: 1px 0;
```

Reply:
Good point. I'll remove these references and just use pixel values.

I've taken another look at the styling on the existing VS Code grid and copied the padding styling over more closely.

Reply:
Turns out the grid itself shouldn't really have any padding. I was accidentally copying this from the `VSCodeDataGridRow` component, but we can't really add padding to rows anymore because each cell is separate. I'm going to just skip this `1px` padding.